src/commands/plugins.rs

```diff
@@ -35,19 +35,58 @@ impl PluginCommands {
 #[derive(Parser, Debug)]
 pub struct Install {
     /// Name of Spin plugin.
-    pub name: String,
-    // If present, updates existing plugins instead of skipping.
-    // TODO: think about how to handle breaking changes
-    // #[structopt(long = "update")]
-    // pub update: bool,
+    #[clap(
+        name = "PLUGIN_NAME",
+        conflicts_with = "REMOTE_PLUGIN_MANIFEST",
+        conflicts_with = "LOCAL_PLUGIN_MANIFEST"
+    )]
+    pub name: Option<String>,
+    /// source of local manifest file
+    #[clap(
+        name = "LOCAL_PLUGIN_MANIFEST",
+        short = 'f',
+        long = "file",
+        conflicts_with = "REMOTE_PLUGIN_MANIFEST"
+    )]
+    pub local_manifest_src: Option<String>,
+    /// source of remote manifest file
+    #[clap(
+        name = "REMOTE_PLUGIN_MANIFEST",
+        short = 'u',
+        long = "url",
+        conflicts_with = "LOCAL_PLUGIN_MANIFEST"
+    )]
+    pub remote_manifest_src: Option<Url>,
+    /// skips prompt to accept the installation of the plugin.
+    #[clap(short = 'y', long = "yes", takes_value = false)]
+    pub skip_accept_install: bool,
+    /// specify particular version of plugin to install from centralized repository
+    #[clap(
+        long = "version",
+        short = 'v',
+        conflicts_with = "REMOTE_PLUGIN_MANIFEST",
+        conflicts_with = "LOCAL_PLUGIN_MANIFEST",
+        requires("PLUGIN_NAME")
+    )]
+    pub version: Option<String>,
 }
 
 impl Install {
     pub async fn run(self) -> Result<()> {
-        println!("The name of the plugin being installed {:?}", self.name);
-        PluginInstaller::new(&self.name, SPIN_PLUGINS_REPO, get_spin_plugins_directory()?)?
-            .install()
-            .await?;
+        println!("Attempting to install plugin: {:?}", self.name);
+        PluginInstaller::new(
+            match &self.name {
+                Some(name) => name,
+                _ => "",
```

Comment:
It feels a little less ergonomic to specify the name when installing from either a local/remote manifest.

`spin plugin install -f <path_to_plugin>` seems more ergonomic and also since we are pointing at the manifest, we will be able to obtain the plugin name from the manifest. The plugin name is only explicitly required when we fetch from the centralized repository to identify which particular manifest needs to be opened.